calendarsubscription/http.go

```diff
@@ -12,7 +12,7 @@ import (
 func (s *Store) ServeICalData(w http.ResponseWriter, req *http.Request) {
 	ctx := req.Context()
 	src := permission.Source(ctx)
-	if src.Type != permission.SourceTypeCalendarSubscription {
+	if src.Type != permission.SourceTypeCalendarSubscription || isCreationDisabled(ctx) {
```

Comment:
Instead of checking this here, do it in the `Authorize` method so that the context is never given the authorization to start.

It also might be clearer to just do the cfg check in both places without a helper since it's a bool check

Reply:
Sounds good

Reply:
Yeah, I made the helper as part of the AC for the story but since it's small enough it would make sense to do it inline where needed